jena-rdfconnection/src/main/java/org/apache/jena/rdfconnection/RDFConnectionRemote.java

```diff
@@ -0,0 +1,478 @@
+import org.apache.jena.riot.* ;
+import org.apache.jena.riot.web.HttpCaptureResponse ;
+import org.apache.jena.riot.web.HttpOp ;
+import org.apache.jena.riot.web.HttpResponseLib ;
+import org.apache.jena.sparql.ARQException ;
+import org.apache.jena.sparql.core.DatasetGraph ;
+import org.apache.jena.sparql.core.Transactional ;
+import org.apache.jena.system.Txn ;
+import org.apache.jena.update.UpdateExecutionFactory ;
+import org.apache.jena.update.UpdateProcessor ;
+import org.apache.jena.update.UpdateRequest ;
+import org.apache.jena.web.HttpSC ;
+
+/** 
+ * 
+ */
+public class RDFConnectionRemote implements RDFConnection {
+    private static final String fusekiDftSrvQuery   = "sparql" ;
+    private static final String fusekiDftSrvUpdate  = "update" ;
+    private static final String fusekiDftSrvGSP     = "data" ;
+    
+    private boolean isOpen = true ; 
+    private final String destination ;
+    private final String svcQuery ;
+    private final String svcUpdate ;
+    private final String svcGraphStore ;
+    private HttpClient httpClient ;
+    private HttpContext httpContext = null ;
+    
+    // Builder?
+    // HttpContext, HttpClient.
+    // Statics for "query", "query+update" : SparqlQueryConnectionRemote > SparqlUpdateConnectionRemote > RDFConnectionRemote
+    // XXX Very long "HttpOp.execHttpPost"
+    
+    /** Create connection that wil use the {@link HttpClient} using URL of the dataset and default service names */
+    public RDFConnectionRemote(HttpClient httpClient, String destination) {
+        this(httpClient,
+             requireNonNull(destination),
+             fusekiDftSrvQuery, 
+             fusekiDftSrvUpdate,
+             fusekiDftSrvGSP) ;
+    }
+
+
+    /** Create connection, using URL of the dataset and default service names */
+    public RDFConnectionRemote(String destination) {
+        this(requireNonNull(destination),
+             fusekiDftSrvQuery, 
+             fusekiDftSrvUpdate,
+             fusekiDftSrvGSP) ;
+    }
+
+    // ??
+    /** Create connection, using full URLs for services. Pass a null for "no service endpoint". */
+    public RDFConnectionRemote(String sQuery, String sUpdate, String sGSP) {
+        this(null, sQuery, sUpdate, sGSP) ;
+    }
+    
+    /** Create connection, using URL of the dataset and short names for the services */
+    public RDFConnectionRemote(String destination, String sQuery, String sUpdate, String sGSP) {
+        this(null, destination, sQuery, sUpdate, sGSP) ;
+    }
+    
+    /** Create connection, using URL of the dataset and short names for the services */
+    public RDFConnectionRemote(HttpClient httpClient, String destination, String sQuery, String sUpdate, String sGSP) {
+        this.destination = destination ;
+        this.svcQuery = formServiceURL(destination,sQuery) ;
+        this.svcUpdate = formServiceURL(destination,sUpdate) ;
+        this.svcGraphStore = formServiceURL(destination,sGSP) ;
+//        if ( httpClient == null )
```

Comment:
I am guessing that the null check is unnecessary because passing a null client to `HttpOp` causes the default client to be used?